examples/seismic/acoustic/wavesolver.py

```diff
@@ -38,21 +42,14 @@ def __init__(self, model, source, receiver,
         # Cache compiler options
         self._kwargs = kwargs
 
-    @cached_property
-    def op_fwd(self):
+    @memoized
+    def op_fwd(self, save=False):
```

Comment:
We probably want to use `@memoized` for all other operator types too, for consistency. Unless I'm missing something, this will also allows us to drop the dependency on `cached_property` completely. 

We probably also want to drop the `save` keyword when calling the memoized operators, since it's not actually used and just swallowed with no effect by `kwargs`.

Reply:
I initially tried `memoized` everywhere, but it seems to not like functions without `kwargs`.

>We probably also want to drop the save keyword when calling the memoized 

Wouldn't it ignore it? We would still need a new operator and generate new code if `save` changes.